src/main/java/com/google/devtools/build/lib/remote/RemoteCache.java

```diff
@@ -348,12 +388,7 @@ public void download(
           tmpOutErr.clearErr();
         }
       } catch (IOException e) {
-        if (downloadException != null && e != downloadException) {
-          e.addSuppressed(downloadException);
-        }
-        if (interruptedException != null) {
-          e.addSuppressed(interruptedException);
-        }
+        downloadException.add(e);
```

Comment:
This seems to be missing initialization for downloadException - how do you know it's not null at this point?

Reply:
Truth. Will fix.

Reply:
Actually, my treatment of this exception is wrong, I'll just redo it.

Reply:
And this was another reason specifically not to have DownloadException be an IOException (despite being badly behaved here) - we cannot possibly be handling the same IOException to be a suppression parent if DownloadException is *not* an IOException (see my previous change to add the `e != downloadException` here).